ui/app/pages/confirm-transaction/conf-tx.js

```diff
@@ -45,188 +40,220 @@ function mapStateToProps (state) {
   }
 }
 
-inherits(ConfirmTxScreen, Component)
-function ConfirmTxScreen () {
-  Component.call(this)
-}
-
-ConfirmTxScreen.prototype.getUnapprovedMessagesTotal = function () {
-  const {
-    unapprovedMsgCount = 0,
-    unapprovedPersonalMsgCount = 0,
-    unapprovedTypedMessagesCount = 0,
-  } = this.props
-
-  return unapprovedTypedMessagesCount + unapprovedMsgCount + unapprovedPersonalMsgCount
-}
-
-ConfirmTxScreen.prototype.componentDidMount = function () {
-  const {
-    unapprovedTxs = {},
-    network,
-    send,
-  } = this.props
-  const unconfTxList = txHelper(unapprovedTxs, {}, {}, {}, network)
-
-  if (unconfTxList.length === 0 && !send.to && this.getUnapprovedMessagesTotal() === 0) {
-    this.props.history.push(DEFAULT_ROUTE)
+class ConfirmTxScreen extends Component {
+  static propTypes = {
+    unapprovedMsgCount: PropTypes.number,
+    unapprovedPersonalMsgCount: PropTypes.number,
+    unapprovedTypedMessagesCount: PropTypes.number,
+    network: PropTypes.number,
+    index: PropTypes.number,
+    unapprovedTxs: PropTypes.array,
+    unapprovedMsgs: PropTypes.array,
+    unapprovedPersonalMsgs: PropTypes.array,
+    unapprovedTypedMessages: PropTypes.array,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        id: PropTypes.number,
+      }),
+    }),
+
+    selectedAddressTxList: PropTypes.array,
+    currentCurrency: PropTypes.string,
+    blockGasLimit: PropTypes.string,
+    conversionRate: PropTypes.number,
+    history: PropTypes.object,
+    accounts: PropTypes.array,
```

Comment:
```suggestion
    accounts: PropTypes.object,
```
Not that it matters, since `accounts` is unused, but this should be an object as well.